packages/api/src/beacon/routes/beacon/rewards.ts

```diff
@@ -0,0 +1,88 @@
+import {ContainerType} from "@chainsafe/ssz";
+import {ssz, ValidatorIndex} from "@lodestar/types";
+
+import {
+  RoutesData,
+  ReturnTypes,
+  Schema,
+  ReqSerializers,
+  ContainerDataExecutionOptimistic,
+} from "../../../utils/index.js";
+import {HttpStatusCode} from "../../../utils/client/httpStatusCode.js";
+import {ApiClientResponse} from "../../../interfaces.js";
+import {BlockId} from "./block.js";
+
+// See /packages/api/src/routes/index.ts for reasoning and instructions to add new routes
+
+/**
+ * True if the response references an unverified execution payload. Optimistic information may be invalidated at
+ * a later time. If the field is not present, assume the False value.
+ */
+export type ExecutionOptimistic = boolean;
+
+export type BlockRewards = {
+  proposerIndex: ValidatorIndex;
+  total: number;
+  attestations: number;
+  syncAggregate: number;
+  proposerSlashings: number;
+  attesterSlashings: number;
+};
+
+export type Api = {
+  /**
+   * Get block rewards
+   * Returns the info of rewards received by the block proposer
+   *
+   * @param blockId Block identifier.
+   * Can be one of: "head" (canonical head in node's view), "genesis", "finalized", \<slot\>, \<hex encoded blockRoot with 0x prefix\>.
+   */
+  getBlockRewards(
+    blockId: BlockId
+  ): Promise<
+    ApiClientResponse<
+      {[HttpStatusCode.OK]: {data: BlockRewards; executionOptimistic: ExecutionOptimistic}},
+      HttpStatusCode.BAD_REQUEST | HttpStatusCode.NOT_FOUND
+    >
+  >;
+};
+
+/**
+ * Define javascript values for each route
+ */
+export const routesData: RoutesData<Api> = {
+  getBlockRewards: {url: "/eth/v1/beacon/rewards/blocks/{block_id}", method: "GET"},
+};
+
+export type ReqTypes = {
+  /* eslint-disable @typescript-eslint/naming-convention */
+  getBlockRewards: {params: {block_id: string}};
+};
+
+export function getReqSerializers(): ReqSerializers<Api, ReqTypes> {
+  return {
+    getBlockRewards: {
+      writeReq: (block_id) => ({params: {block_id: String(block_id)}}),
+      parseReq: ({params}) => [params.block_id],
+      schema: {params: {block_id: Schema.StringRequired}},
+    },
+  };
+}
+
+export function getReturnTypes(): ReturnTypes<Api> {
+  const BlockRewardsResponse = new ContainerType(
+    {
+      proposerIndex: ssz.ValidatorIndex,
+      total: ssz.UintNum64,
+      attestations: ssz.UintNum64,
```

Comment:
> Considering Number.MAX_SAFE_INTEGER is 9007199254740991 (16 digits) we should be fine if we use gwei but not if we would use wei
>
>I didn't check, there might have been already discussions on this when we implemented the rewards cache.

Yea all rewards values except for `produceBlockV3` are in Gwei. 
The choice of choosing `UintNum64` here and the rewards cache as oppose to `Gwei(UintBn64)` is exactly what you described: there is no need to use BigInt here because there is no way the reward value would go as high as `Number.MAX_SAFE_INTEGER` in Gwei.

If we want to be semantic, we can define a `sszType` named `SmallGwei` or something which is an alias of `UintNum64`, but I don't see the urgent need for it.